### PR TITLE
Enhance common module code documentation

### PR DESCRIPTION
**Pull request by Ygnas**
# Issue link

[https://issues.redhat.com/browse/RHOAIENG-13958](https://issues.redhat.com/browse/RHOAIENG-13958)

# What changes have been made

Enhanced common module code documentation
# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [x] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk/common/utils/demos.py [hunk 1]
```diff
@@ -13,10 +13,14 @@ def copy_demo_nbs(dir: str = "./demo-notebooks", overwrite: bool = False):
     Any files that exist in the directory that don't match these values will remain untouched.
 
     Args:
-        dir (str): The directory to copy the demo notebooks to. Defaults to "./demo-notebooks". overwrite (bool):
-        overwrite (bool): Whether to overwrite files in the directory if it already exists. Defaults to False.
+        dir (str):
+            The directory to copy the demo notebooks to. Defaults to "./demo-notebooks". overwrite (bool):
```
Comment (Bobbins228) [inline src/codeflare_sdk/common/utils/demos.py:17]:
```suggestion
            The directory to copy the demo notebooks to. Defaults to "./demo-notebooks". 
```

**Review by Bobbins228 — CHANGES_REQUESTED** [review]

**Review by Bobbins228 — COMMENTED** [review]
/lgtm

**Comment by ChristianZaccaria** [comment]
I'm making a few changes in this repo's settings:
- We are dropping the merge queue feature as it seems it's not properly maintained with major issues found unresolved for over a year.
- One benefit seen from merge queue is that it ensures PRs are tested against the target branch. Since we won't be using this feature, I've enabled "Require branches to be up to date before merging" GitHub feature in this repo. By enabling this, all PRs will require to rebase before they can be merged and this way ensure code stability and integration.

**Review by ChristianZaccaria — COMMENTED** [review]
Great work! 

Just to confirm, do we want to cover code documentation on only the public functions in the common module?

**Comment by Ygnas** [comment]
> Great work!
> 
> Just to confirm, do we want to cover code documentation on only the public functions in the common module?

It could cover the private ones as well, but I don't see any private functions appear in the sphinx documentation.

**Comment by ChristianZaccaria** [comment]
@Ygnas Yeah that's what I thought as well. Perhaps, the private functions can continue to have _simple comments_ while the public functions include the enhanced code documentation for sphynx.

**Review by ChristianZaccaria — COMMENTED** [review]
/lgtm

File: src/codeflare_sdk/common/kubernetes_cluster/auth.py [hunk 2]
```diff
@@ -165,8 +165,18 @@ def load_kube_config(self):
 
 def config_check() -> str:
     """
-    Function for loading the config file at the default config location ~/.kube/config if the user has not
-    specified their own config file or has logged in with their token and server.
+    Check and load the Kubernetes config from the default location.
+
+    This function checks if a Kubernetes config file exists at the default path
```
Comment (KPostOffice) [inline src/codeflare_sdk/common/kubernetes_cluster/auth.py:170]: It also loads from the global variable from auth.py if it's set.

File: src/codeflare_sdk/common/kueue/kueue.py [hunk 3]
```diff
@@ -19,8 +19,21 @@
...
 
 
-def get_default_kueue_name(namespace: str):
-    # If the local queue is set, use it. Otherwise, try to use the default queue.
+def get_default_kueue_name(namespace: str) -> Optional[str]:
+    """
+    Retrieves the default Kueue name from the provided namespace.
+
+    This function attempts to fetch the local queues in the given namespace and checks if any of them is annotated
+    as the default queue. If found, the name of the default queue is returned.
```
Comment (KPostOffice) [inline src/codeflare_sdk/common/kueue/kueue.py:27]: add a small comment on the actual label that marks the LQ which marks the queue as the default queue here `kueue.x-k8s.io/default-queue`

**Review by KPostOffice — COMMENTED** [review]
looks good, a couple of comments

**Review by KPostOffice — APPROVED** [review]
/lgtm

**Review by Bobbins228 — APPROVED** [review]
/lgtm /approve

**Pull request merged**